qiskit_ibm_runtime/sampler.py

```diff
@@ -180,14 +181,25 @@ def run(
         Raises:
             ValueError: If the input values are invalid.
         """
-        if isinstance(circuits, Iterable) and not all(
-            isinstance(inst, QuantumCircuit) for inst in circuits
-        ):
-            raise ValueError(
-                "The circuits parameter has to be instances of QuantumCircuit."
-            )
+        if isinstance(circuits, QuantumCircuit):
+            circuits = [circuits]
+
+        for circ in circuits:
+            if not isinstance(circ, QuantumCircuit):
+                raise ValueError(
+                    "The circuits parameter has to be instances of QuantumCircuit."
+                )
+            if circ.num_clbits == 0:
+                raise ValueError("The circuits should have at least one classical bit.")
+            if self.options.resilience_level >= 1 and circ.num_clbits > len(
```

Comment:
Is `>` enough to check or do we need to check `!=` for the lengths?

Reply:
Yes, it's impossible to set more measurements than classical bits in the final measurements (duplicate measurements are removed).

Reply:
But, it's clear to use `!=`. So, I update it.